lib/src/utils/file_cache.dart

```diff
@@ -14,10 +14,22 @@ class FileCache {
   List<int> operator [](Object filePath) {
     assert(filePath is String);
     if (_activeFiles.containsKey(filePath)) return _activeFiles[filePath];
-    return new File(filePath).readAsLinesSync().map((l) => l.length).toList();
+    // Don't use readAsLines() or we'll lose track of the different line endings (\r\n).
+    return splitIntoLines(new File(filePath).readAsStringSync());
   }
 
   void operator []=(String filePath, List<int> lines) {
     _activeFiles[filePath] = lines;
   }
 }
+
+// TODO: Where should this live?
```

Comment:
There is already a method `findLineLengths` in `position_convert.dart` - we should update that to use this logic.

Reply:
This PR already updates the functions in there. This comment was about whether this function `splitIntoLines` would be better somewhere else. Previously the tests and real code were splitting the lines themselves, but since the logic isn't straight forward now I created this function for them. It doesn't feel like it should be in `file_cache.dart`?

Reply:
It looks like `positionFromOffset` is updated but not `findLineLengths` on line 43 in `position_convert.dart`

Reply:
Oh whoops, I totally didn't read properly! I'll fix :-)

Reply:
I've moved the implementation into `findLineLengths` and deleted `splitIntoLines` since it was an exact dupe :)